src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -177,58 +177,80 @@ public static synchronized ArrayList getFifoNextSpanData(
 
-    // Run string EBD export call if enabled
-    if (stringHistoryEnabled) {
-      final String ebdStringFileName =
+      // Run standard EBD export call (int, float, ...)
+      final String ebdFileName =
           HistoricalDataConstants.QUEUE_FILE_FOLDER
               + "/"
-              + HistoricalDataConstants.QUEUE_EBD_STRING_FILE_NAME
+              + HistoricalDataConstants.QUEUE_EBD_FILE_NAME
               + HistoricalDataConstants.QUEUE_FILE_EXTENSION;
-      stringHistorical = true;
+      boolean stringHistorical = false;
       HistoricalDataManager.exportHistoricalToFile(
           ebdStartTime,
           ebdEndTime,
-          ebdStringFileName,
+          ebdFileName,
           includeTagGroupA,
           includeTagGroupB,
           includeTagGroupC,
           includeTagGroupD,
           stringHistorical);
 
-      // Parse string EBD export call and combine with standard EBD call results
-      ArrayList queueStringData = HistoricalDataManager.parseHistoricalFile(ebdStringFileName);
-      queueData.addAll(queueStringData);
-    }
+      // Parse standard EBD export call
+      queueData = HistoricalDataManager.parseHistoricalFile(ebdFileName);
 
-    // Store end time +1 ms (to prevent duplicate data)
-    final String newTimeTrackerVal = Long.toString(endTimeTrackerMsLong + 1);
-    FileAccessManager.writeStringToFile(timeMarkerFileName, newTimeTrackerVal);
+      // Run string EBD export call if enabled
+      if (stringHistoryEnabled) {
+        final String ebdStringFileName =
+            HistoricalDataConstants.QUEUE_FILE_FOLDER
+                + "/"
+                + HistoricalDataConstants.QUEUE_EBD_STRING_FILE_NAME
+                + HistoricalDataConstants.QUEUE_FILE_EXTENSION;
+        stringHistorical = true;
+        HistoricalDataManager.exportHistoricalToFile(
+            ebdStartTime,
+            ebdEndTime,
+            ebdStringFileName,
+            includeTagGroupA,
+            includeTagGroupB,
+            includeTagGroupC,
+            includeTagGroupD,
+            stringHistorical);
+
+        // Parse string EBD export call and combine with standard EBD call results
+        ArrayList queueStringData = HistoricalDataManager.parseHistoricalFile(ebdStringFileName);
+        queueData.addAll(queueStringData);
+      }
+
+      // Store end time +1 ms (to prevent duplicate data)
+      final String newTimeTrackerVal = Long.toString(endTimeTrackerMsLong + 1);
+      FileAccessManager.writeStringToFile(timeMarkerFileName, newTimeTrackerVal);
+    }
 
     // Return data
     return queueData;
   }
+
+  /**
+   * Compare a start and end time in milliseconds to see if they are in the same second.
+   *
+   * @param smallerTimeMillis the start time in milliseconds.
+   * @param biggerTimeMillis the end time in milliseconds.
+   * @return true if the times are in the same second
+   */
+  private static boolean sameSecond(long smallerTimeMillis, long biggerTimeMillis) {
```

Comment:
This logic seems flawed as 1609787800999 and 1609787801000 would be grouped in the same second but they are not actually the same second

Reply:
is what you wanted to do 

if ((num1/1000) == (num2/1000)){}?

Reply:
yeah this needs to be changes, will repush

Reply:
okay